yt_dlp/extractor/twitch.py

```diff
@@ -407,6 +442,20 @@ def _extract_moments(self, info, item_id):
                 'title': name,
             }
 
+    # Logical chapters
+    #   if category didn't change, the only chapter is the category, spanning the whole stream
+    def _extract_chapters(self, info, item_id):
+        chapters = list(self._extract_moments(info, item_id))
+
+        if len(chapters) == 0:
+            chapters.append({
+                'start_time': 0,
+                'end_time': info.get('lengthSeconds'),
+                'title': traverse_obj(info, ('game', 'displayName'))
+            })
+
+        return chapters
+
```

Comment:
1. Moments+chapters should be a single function
2. Due to some core improvements, the endtime calculation is now unnecessary
3. The comments are not needed

```py
    def _extract_chapters(self, info, item_id):
        if not info.get('moments'):
            game = traverse_obj(info, ('game', 'displayName'))
            if game:
                yield {'title': game}
            return
            
        for moment in info.get('moments'):
            ...
```